src/InnoSetup/PreInstall.iss

```diff
@@ -431,7 +431,11 @@ begin
 
   if not IDFUseExisting then
   begin
-    IDFAddDownload();
+    if WildCardMatch(IDFDownloadVersion, 'v*') then
```

Comment:
Now that you aren't calling this function for branches, a lot of code can be removed from IDFAddDownload:
https://github.com/espressif/idf-installer/blob/1ae9e300475f332fa1045708b5298e78c4bd28d3/src/InnoSetup/Environment.iss#L176-L196